FirebaseAuth/Tests/Unit/FIRAuthTests.m

```diff
@@ -1805,6 +1849,181 @@ - (void)testSignInAndRetrieveDataWithCustomTokenFailure {
 
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_OSX || TARGET_OS_MACCATALYST
+/** @fn testStartPasskeySignInSuccess
+    @brief Tests the flow of a successful @c startPasskeySignInWithCompletion: call.
+ */
+- (void)testStartPasskeySignInSuccess {
+  if (@available(iOS 15.0, tvOS 16.0, macOS 12.0, *)) {
+    OCMExpect([_mockBackend startPasskeySignIn:[OCMArg any] callback:[OCMArg any]])
+        .andCallBlock2(^(FIRStartPasskeySignInRequest *_Nullable request,
+                         FIRStartPasskeySignInResponseCallback callback) {
+          XCTAssertEqualObjects(request.APIKey, kAPIKey);
+          dispatch_async(FIRAuthGlobalWorkQueue(), ^() {
+            id mockStartPasskeySignInResponse = OCMClassMock([FIRStartPasskeySignInResponse class]);
+            OCMStub([mockStartPasskeySignInResponse rpID]).andReturn(kRpId);
+            OCMStub([mockStartPasskeySignInResponse challenge]).andReturn(kChallenge);
+            callback(mockStartPasskeySignInResponse, nil);
+          });
+        });
+    XCTestExpectation *expectation = [self expectationWithDescription:@"callback"];
+    [[FIRAuth auth] signOut:NULL];
+    [[FIRAuth auth]
+        startPasskeySignInWithCompletion:^(
+            ASAuthorizationPlatformPublicKeyCredentialAssertionRequest *_Nullable request,
+            NSError *_Nullable error) {
+          XCTAssertNil(error);
+          XCTAssertEqualObjects([[request challenge] base64EncodedStringWithOptions:0], kChallenge);
+          ASAuthorizationPlatformPublicKeyCredentialProvider *provider =
+              (ASAuthorizationPlatformPublicKeyCredentialProvider *)[request provider];
+          XCTAssertEqualObjects([provider relyingPartyIdentifier], kRpId);
+          [expectation fulfill];
+        }];
+
+    [self waitForExpectationsWithTimeout:kExpectationTimeout handler:nil];
+    OCMVerifyAll(_mockBackend);
+  }
+}
+
+/** @fn testStartPasskeySignInFailure
+    @brief Tests the flow of a failed @c startPasskeySignInWithCompletion: call.
+ */
+- (void)testStartPasskeySignInFailure {
+  if (@available(iOS 15.0, tvOS 16.0, macOS 12.0, *)) {
+    OCMExpect([_mockBackend startPasskeySignIn:[OCMArg any] callback:[OCMArg any]])
+        .andDispatchError2([FIRAuthErrorUtils operationNotAllowedErrorWithMessage:nil]);
+    XCTestExpectation *expectation = [self expectationWithDescription:@"callback"];
+    [[FIRAuth auth] signOut:NULL];
+    [[FIRAuth auth]
+        startPasskeySignInWithCompletion:^(
+            ASAuthorizationPlatformPublicKeyCredentialAssertionRequest *_Nullable request,
+            NSError *_Nullable error) {
+          XCTAssertNil(request);
+          XCTAssertEqual(error.code, FIRAuthErrorCodeOperationNotAllowed);
+          XCTAssertNotNil(error.userInfo[NSLocalizedDescriptionKey]);
+          [expectation fulfill];
+        }];
+    [self waitForExpectationsWithTimeout:kExpectationTimeout handler:nil];
+    OCMVerifyAll(_mockBackend);
+  }
+}
+
+// TODO @liubin uncomment this after expires_in changes
```

Comment:
We can include this in a future pr

Reply:
Got it, removed for now.  Will add it back with the "expires_in" fix PR.